### PR TITLE
``stack`` raises error when stacking signals that contain a variance signal

### PR DESCRIPTION
**Pull request by francisco-dlp**
### Progress of the PR
- [x] Change implemented (can be split into several points),
- [x] add an changelog entry in the `upcoming_changes` folder (see [`upcoming_changes/README.rst`](https://github.com/hyperspy/hyperspy/blob/RELEASE_next_minor/upcoming_changes/README.rst)),
- [x] Check formatting changelog entry in the `readthedocs` doc build of this PR (link in github checks)
- [x] add tests,
- [x] ready for review.

### Minimal example of the bug fix or the new feature

The following raises an error without this fix:

```python
import hyperspy.api as hs
import numpy as np

s = hs.signals.EELSSpectrum(np.random.random((10, 10, 100)))
s.axes_manager[0].name = "x"
s.axes_manager[1].name = "y"
s.estimate_poissonian_noise_variance()
hs.stack([s, s], axis="y")
```

### CODE REVIEW

File: hyperspy/axes.py [hunk 1]
```diff
@@ -1540,8 +1540,11 @@ def __getitem__(self, y):
         return ans
 
     def _axes_getter(self, y):
-        if y in self._axes:
-            return y
+        if isinstance(y, BaseDataAxis):
+            if y in self._axes:
+                return y
+            else:
+                raise ValueError(f"{y} is not in {self}")
```
Comment (jlaehne) [inline hyperspy/axes.py:1547]: CAn you get this line covered by a test?

**Review by jlaehne — COMMENTED** [review]
otherwise, lgtm

**Review by ericpre — APPROVED** [review]
Other than adding a test, as mentioned by @jlaehne, it looks good!

File: hyperspy/axes.py (hunk 1 above)
Comment (francisco-dlp) [reply hyperspy/axes.py:1547]: Done.

**Pull request merged**